src/firefly/java/edu/caltech/ipac/astro/net/HorizonsEphPairs.java

```diff
@@ -58,9 +58,20 @@ public static HorizonsResults[] lowlevelGetEphInfo(String idOrName) throws Faile
 
         try {
             String urlStr = horizonsServer + path + "?sstr="+ URLEncoder.encode(idOrName, "UTF-8");
-            URL url = new URL(urlStr);
-            String jsonStrResult = URLDownload.getDataFromURL(url, null,null,null).getResultAsString();
-            JsonHelper json= JsonHelper.parse(jsonStrResult);
+            var result = URLDownload.getDataFromURL(new URL(urlStr), null,null,null);
+            if (result.getResponseCode() != 200) {
+                throw new FailedRequestException("failed to retrieve ephemeris pairs, response code: " + result.getResponseCode());
+            }
+            String jsonStrResult = result.getResultAsString();
+//            test using HttpServices
+//            OutputStream resultStream = new ByteArrayOutputStream();
+//            HttpServices.Status stat= HttpServices.getData(urlStr, resultStream);
+//            if (stat.isError()) {
+//                throw new FailedRequestException("failed to retrieve ephemeris pairs, response code: " + stat.getStatusCode());
+//            }
+//            String jsonStrResult = resultStream.toString();
```

Comment:
did you intend to remove this code or keep it for testing purposes?

Reply:
I should remove it now